From 921c68e061fff3baf787275a1c1a446a02de8ec3 Mon Sep 17 00:00:00 2001
From: Pete Skeggs <plskeggs@noeticdesign.com>
Date: Sun, 14 Jun 2026 20:50:23 -0700
Subject: [PATCH] Go back to previous method

The recent change is not compatible
with the inconsistent cobids between
the battery card firmware and the
mag card. The battery card was not
already pre-adding the nodeId, but the
mag card was.

Instead, we'll go back to this,
and then change the mag card to not
use pre-added node ids.
---
 stack/CO_PDO.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/stack/CO_PDO.c b/stack/CO_PDO.c
index 1696d39..3fcdb7f 100644
--- a/stack/CO_PDO.c
+++ b/stack/CO_PDO.c
@@ -149,7 +149,11 @@ static void CO_TPDOconfigCom(CO_TPDO_t* TPDO, uint32_t COB_IDUsedByTPDO, uint8_t
     /* is TPDO used? */
     if((COB_IDUsedByTPDO & 0xBFFFF800L) == 0 && TPDO->dataLength && ID){
         /* is used default COB-ID? */
-        if(ID == TPDO->defaultCOB_ID) ID += TPDO->nodeId;
+        /* Oresat: we have 2 cards with 10 TPDOs, so this change is important.
+           Comment out this if statement:
+           if(ID == TPDO->defaultCOB_ID)
+         */
+        ID += TPDO->nodeId;
         TPDO->valid = true;
     }
     else{
-- 
2.43.0

